include/loads_bcs/constraints.h

```diff
@@ -91,6 +124,10 @@ class Constraints {
   std::shared_ptr<mpm::Mesh<Tdim>> mesh_;
   //! Logger
   std::unique_ptr<spdlog::logger> console_;
+  //! List of absorbing constraint ptrs
+  std::vector<std::shared_ptr<mpm::AbsorbingConstraint>> absorbing_constraint_;
+  //! List of absorbing constraint node set Ids
+  std::vector<unsigned> absorbing_nset_id_;
```

Comment:
This looks better. I have some questions though. Let's discuss this in person tomorrow.

Reply:
What are your questions?